tests/CMakeLists.txt

```diff
@@ -74,10 +74,18 @@ add_gtest(decoder decoder.cpp)
 
 add_gtest(client client.cpp)
 # We're mocking the mg_secure_transport_init function in the test.
-target_link_libraries(client -Wl,--wrap=mg_secure_transport_init)
+if(MGCLIENT_ON_APPLE)
+    target_link_libraries(client -Wl,-alias,___wrap_mg_secure_transport_init,_mg_secure_transport_init)
```

Comment:
I was thinking about using only this on all platforms, but that's not possible because symbol names are not the same across platforms. E.g. ON_LINUX `mg_secure_transport_init` is `mg_secure_transport_init`, while ON_APPLE is `_mg_secure_transport_init`.